src/actions/GridActions.js

```diff
@@ -238,7 +238,11 @@ export const doRemoteSort = ({
         );
 
         if (typeof dataSource === 'function') {
-            return dataSource({pageIndex}, {}, sortParams).then((response) => {
+            return dataSource(
+                {pageIndex},
+                filterFields,
+                sortParams
```

Comment:
This will have to be a breaking change since to either doRemoteSort or setPageIndexAsync and it seems like doRemoteSort will likely have the smallest impact and the pattern for setPageIndexAsync seems more self explanatory.